### ISSUE TITLE
关于CRNN 识别模型在 icdar2015 数据集训练问题

### ISSUE THREAD

**Issue by cobbler1234** [PaddlePaddle/PaddleOCR#1001]
我使用linux CPU样机（没有GPU卡），按照PaddleOCR/doc/doc_ch/recognition.md文档，在 icdar2015数据集上进行训练重现，发现有点过拟合，请问这个结果是否合理？

2020-10-24 07:11:24,951-INFO: Test iter: 15500, acc:0.397689, best_acc:0.420799, best_epoch:31, best_batch_id:500, eval_sample_num:2077
2020-10-24 07:12:28,483-INFO: epoch: 969, iter: 15510, lr: 0.000060, 'loss': 2.463798, 'acc': 1.0, time: 6.253
2020-10-24 07:13:31,969-INFO: epoch: 970, iter: 15520, lr: 0.000059, 'loss': 2.19085, 'acc': 1.0, time: 7.216
2020-10-24 07:14:34,486-INFO: epoch: 970, iter: 15530, lr: 0.000059, 'loss': 1.586413, 'acc': 1.0, time: 6.274
2020-10-24 07:15:37,874-INFO: epoch: 971, iter: 15540, lr: 0.000059, 'loss': 1.931966, 'acc': 1.0, time: 6.222
2020-10-24 07:16:40,312-INFO: epoch: 971, iter: 15550, lr: 0.000059, 'loss': 2.304559, 'acc': 1.0, time: 6.244
2020-10-24 07:17:43,862-INFO: epoch: 972, iter: 15560, lr: 0.000058, 'loss': 1.958434, 'acc': 1.0, time: 6.236
2020-10-24 07:18:47,366-INFO: epoch: 973, iter: 15570, lr: 0.000058, 'loss': 1.779439, 'acc': 1.0, time: 6.236
2020-10-24 07:19:50,579-INFO: epoch: 973, iter: 15580, lr: 0.000058, 'loss': 1.915732, 'acc': 1.0, time: 6.350

**Comment by LDOUBLEV** [comment]
不合理，过拟合的太严重了，验证集精度呢？

**Comment by cobbler1234** [comment]
测试集的精度只有40%左右
2020-10-24 07:11:24,951-INFO: Test iter: 15500, acc:0.397689, best_acc:0.420799, best_epoch:31, best_batch_id:500, eval_sample_num:2077

**Comment by cobbler1234** [comment]
训练log的最开始部分，如下：
grep: warning: GREP_OPTIONS is deprecated; please use an alias or script
2020-10-23 03:16:16,921-INFO: {'TestReader': {'reader_function': 'ppocr.data.rec.dataset_traversal,SimpleReader'}, 'EvalReader': {'label_file_path': './train_data/ic15_data/rec_gt_test.txt', 'img_set_dir': './train_data/ic15_data', 'reader_function': 'ppocr.data.rec.dataset_traversal,SimpleReader'}, 'Architecture': {'function': 'ppocr.modeling.architectures.rec_model,RecModel'}, 'TrainReader': {'label_file_path': './train_data/ic15_data/rec_gt_train.txt', 'num_workers': 8, 'img_set_dir': './train_data/ic15_data', 'reader_function': 'ppocr.data.rec.dataset_traversal,SimpleReader'}, 'Loss': {'function': 'ppocr.modeling.losses.rec_ctc_loss,CTCLoss'}, 'Optimizer': {'function': 'ppocr.optimizer,AdamDecay', 'beta1': 0.9, 'decay': {'total_epoch': 1000, 'step_each_epoch': 20, 'function': 'cosine_decay'}, 'beta2': 0.999, 'base_lr': 0.0005}, 'Backbone': {'function': 'ppocr.modeling.backbones.rec_mobilenet_v3,MobileNetV3', 'scale': 0.5, 'model_name': 'large'}, 'Global': {'algorithm': 'CRNN', 'debug': False, 'max_text_length': 25, 'log_smooth_window': 20, 'save_inference_dir': None, 'character_type': 'en', 'distort': True, 'pretrain_weights': './pretrain_models/rec_mv3_none_bilstm_ctc/best_accuracy', 'image_shape': [3, 32, 100], 'test_batch_size_per_card': 256, 'loss_type': 'ctc', 'use_gpu': False, 'checkpoints': None, 'eval_batch_step': 500, 'print_batch_step': 10, 'save_epoch_step': 300, 'reader_yml': './configs/rec/rec_icdar15_reader.yml', 'epoch_num': 1000, 'save_model_dir': './output/rec_CRNN', 'infer_img': None, 'train_batch_size_per_card': 256}, 'Head': {'encoder_type': 'rnn', 'SeqRNN': {'hidden_size': 96}, 'function': 'ppocr.modeling.heads.rec_ctc_head,CTCPredict'}}
2020-10-23 03:16:17,327-INFO: If regularizer of a Parameter has been set by 'fluid.ParamAttr' or 'fluid.WeightNormParamAttr' already. The Regularization[L2Decay, regularization_coeff=0.000000] in Optimizer will not take effect, and it will only be applied to other Parameters!
2020-10-23 03:16:18,465-INFO: Distort operation can only support in GPU.Distort will be set to False.
import ujson error: No module named 'ujson' use json
2020-10-23 03:16:18,466-INFO: places would be ommited when DataLoader is not iterable
2020-10-23 03:16:18,466-INFO: Distort operation can only support in GPU.Distort will be set to False.
2020-10-23 03:16:18,559-INFO: Loading parameters from ./pretrain_models/rec_mv3_none_bilstm_ctc/best_accuracy...
2020-10-23 03:16:18,664-INFO: Finish initing model from ./pretrain_models/rec_mv3_none_bilstm_ctc/best_accuracy
W1023 03:16:18.687475 12893 build_strategy.cc:170] fusion_group is not enabled for Windows/MacOS now, and only effective when running with CUDA GPU.
!!! The CPU_NUM is not specified, you should set CPU_NUM in the environment variable list.
CPU_NUM indicates that how many CPUPlace are used in the current task.
And if this parameter are set as N (equal to the number of physical CPU core) the program may be faster.

export CPU_NUM=8 # for example, set CPU_NUM as number of physical CPU core which is 8.

!!! The default number of CPU_NUM=1.
2020-10-23 03:17:28,198-INFO: epoch: 0, iter: 10, lr: 0.000500, 'loss': 1392.426, 'acc': 0.515625, time: 6.238
2020-10-23 03:18:31,622-INFO: epoch: 1, iter: 20, lr: 0.000500, 'loss': 1259.9202, 'acc': 0.548828, time: 6.269
2020-10-23 03:19:34,086-INFO: epoch: 1, iter: 30, lr: 0.000500, 'loss': 1189.4807, 'acc': 0.570312, time: 6.233
2020-10-23 03:20:37,463-INFO: epoch: 2, iter: 40, lr: 0.000500, 'loss': 1114.364, 'acc': 0.570312, time: 6.242
2020-10-23 03:21:41,175-INFO: epoch: 3, iter: 50, lr: 0.000500, 'loss': 1058.4438, 'acc': 0.582031, time: 6.286
2020-10-23 03:22:44,067-INFO: epoch: 3, iter: 60, lr: 0.000500, 'loss': 1004.0737, 'acc': 0.591797, time: 6.272
2020-10-23 03:23:47,972-INFO: epoch: 4, iter: 70, lr: 0.000500, 'loss': 980.4745, 'acc': 0.587891, time: 6.276
2020-10-23 03:24:51,887-INFO: epoch: 5, iter: 80, lr: 0.000500, 'loss': 1065.1083, 'acc': 0.580078, time: 7.227
2020-10-23 03:25:54,963-INFO: epoch: 5, iter: 90, lr: 0.000500, 'loss': 1076.1538, 'acc': 0.574219, time: 6.290
2020-10-23 03:26:59,017-INFO: epoch: 6, iter: 100, lr: 0.000500, 'loss': 1064.0168, 'acc': 0.568359, time: 6.292
2020-10-23 03:28:02,074-INFO: epoch: 6, iter: 110, lr: 0.000500, 'loss': 954.3742, 'acc': 0.589844, time: 6.332
2020-10-23 03:29:06,069-INFO: epoch: 7, iter: 120, lr: 0.000500, 'loss': 945.8047, 'acc': 0.589844, time: 6.307
2020-10-23 03:30:10,028-INFO: epoch: 8, iter: 130, lr: 0.000500, 'loss': 991.6068, 'acc': 0.583984, time: 6.314
2020-10-23 03:31:13,053-INFO: epoch: 8, iter: 140, lr: 0.000500, 'loss': 1012.2839, 'acc': 0.572266, time: 6.284
2020-10-23 03:32:16,773-INFO: epoch: 9, iter: 150, lr: 0.000500, 'loss': 1052.9985, 'acc': 0.550781, time: 6.279
2020-10-23 03:33:20,563-INFO: epoch: 10, iter: 160, lr: 0.000500, 'loss': 1157.7627, 'acc': 0.521484, time: 7.213
2020-10-23 03:34:23,359-INFO: epoch: 10, iter: 170, lr: 0.000500, 'loss': 1074.262, 'acc': 0.533203, time: 6.267
2020-10-23 03:35:27,154-INFO: epoch: 11, iter: 180, lr: 0.000500, 'loss': 1036.8999, 'acc': 0.535156, time: 6.276
2020-10-23 03:36:29,980-INFO: epoch: 11, iter: 190, lr: 0.000500, 'loss': 1072.1671, 'acc': 0.503906, time: 6.290
2020-10-23 03:37:33,734-INFO: epoch: 12, iter: 200, lr: 0.000500, 'loss': 1090.55, 'acc': 0.498047, time: 6.291
2020-10-23 03:38:37,564-INFO: epoch: 13, iter: 210, lr: 0.000500, 'loss': 1090.3582, 'acc': 0.486328, time: 6.272
2020-10-23 03:39:40,509-INFO: epoch: 13, iter: 220, lr: 0.000500, 'loss': 1071.4243, 'acc': 0.466797, time: 6.316
2020-10-23 03:40:44,291-INFO: epoch: 14, iter: 230, lr: 0.000500, 'loss': 1075.75, 'acc': 0.457031, time: 6.290
2020-10-23 03:41:48,239-INFO: epoch: 15, iter: 240, lr: 0.000500, 'loss': 1140.7704, 'acc': 0.4375, time: 7.202
2020-10-23 03:42:51,176-INFO: epoch: 15, iter: 250, lr: 0.000500, 'loss': 1218.2217, 'acc': 0.40625, time: 6.311
2020-10-23 03:43:55,045-INFO: epoch: 16, iter: 260, lr: 0.000500, 'loss': 1210.5253, 'acc': 0.398438, time: 6.289
2020-10-23 03:44:57,910-INFO: epoch: 16, iter: 270, lr: 0.000500, 'loss': 1186.5344, 'acc': 0.400391, time: 6.259
2020-10-23 03:46:01,785-INFO: epoch: 17, iter: 280, lr: 0.000500, 'loss': 1202.8792, 'acc': 0.40625, time: 6.279
2020-10-23 03:47:05,617-INFO: epoch: 18, iter: 290, lr: 0.000500, 'loss': 1178.3191, 'acc': 0.433594, time: 6.316
2020-10-23 03:48:08,601-INFO: epoch: 18, iter: 300, lr: 0.000500, 'loss': 1161.7654, 'acc': 0.439453, time: 6.281
2020-10-23 03:49:12,443-INFO: epoch: 19, iter: 310, lr: 0.000500, 'loss': 1114.1011, 'acc': 0.4375, time: 6.277
2020-10-23 03:50:16,250-INFO: epoch: 20, iter: 320, lr: 0.000500, 'loss': 1113.6431, 'acc': 0.445312, time: 7.212
2020-10-23 03:51:18,985-INFO: epoch: 20, iter: 330, lr: 0.000500, 'loss': 1099.0587, 'acc': 0.458984, time: 6.235
2020-10-23 03:52:22,551-INFO: epoch: 21, iter: 340, lr: 0.000500, 'loss': 1060.1592, 'acc': 0.480469, time: 6.256

**Comment by cobbler1234** [comment]
是不是需要配置正则化相关参数呀？ 我是按照文档默认参数来实验的。

**Comment by LDOUBLEV** [comment]
你不加载了我们的预训练模型训练看下结果

**Comment by cobbler1234** [comment]
不加载预训练模型，似乎效果更差，训练831 epoch，训练集精度已经0.902344，但是测试集精度还是非常低0.002889
2020-10-27 23:50:33,296-INFO: Test iter: 13000, acc:0.002407, best_acc:0.002889, best_epoch:375, best_batch_id:6000, eval_sample_num:2077
2020-10-27 23:51:37,778-INFO: epoch: 813, iter: 13010, lr: 0.000137, 'loss': 153.2096, 'acc': 0.900391, time: 6.303

**Comment by tink2123** [comment]
ic15 的案例是一个demo，因为训练集较小，用默认配置的去训练很容易过拟合。可以尝试增加一些数据增强，另外加正则化也会有帮助。

**Comment by 666888985** [comment]
> 
> 
> 不加载预训练模型，似乎效果更差，训练831 epoch，训练集精度已经0.902344，但是测试集精度还是非常低0.002889
> 2020-10-27 23:50:33,296-INFO: Test iter: 13000, acc:0.002407, best_acc:0.002889, best_epoch:375, best_batch_id:6000, eval_sample_num:2077
> 2020-10-27 23:51:37,778-INFO: epoch: 813, iter: 13010, lr: 0.000137, 'loss': 153.2096, 'acc': 0.900391, time: 6.303

你知道怎么评估自己的数据集吗，我看评估代码eval.py里面是英文的公开数据集

**Comment by sunshineInmoon** [comment]
@cobbler1234 您好，我现在遇到了和您相似的问题，请问您当时解决了吗？

**Comment by sunshineInmoon** [comment]
@cobbler1234 您好，我现在遇到了和您相似的问题，请问您当时解决了吗？
